### ISSUE TITLE
include **Flock** and **Convoy** as a class

### ISSUE THREAD

**Issue by dicaearchus** [movingpandas/movingpandas#322]
#### Is your feature request related to a problem?

A wealth of analysis of trajectories data is present, and these can be grouped into individual metrics (like length, detour). But features do also exist for sections of a trajectory collection. If these are grouped together, this is called a **Flock**. If they follow the same track, but at different revisit times, such subgroups are called a **Convoy**. Such configurations have their own metrics and procedures and it would be nice to have these in moving pandas as well, see for an overview: https://link.springer.com/chapter/10.1007/978-3-319-56759-4_18

#### Describe the solution you'd like
Ideally, it would be nice to have separate classes, as for the **Flock** the timestamps should all be the same. Similarly, for the **Convoy** the trajectory collection should have a trajectory without timestamp, that follows the way the trajectories are following.

#### Describe alternatives you've considered

The **TrajectoryCollection** itself is to open, as any type of trajectory can be included.

#### Additional context

I think including these classes expands the type of analysis that can be made with moving pandas. For example, for the **Convoy**-classes great datasets can be analysed like: https://ops.fhwa.dot.gov/trafficanalysistools/ngsim.htm 
Furthermore, the **TrajectoryAggregator** can be used as a baseline for the **Convoy** class, enhancing use of different classes within movingpandas!

**Comment by bamacgabhann** [comment]
I considered some ideas relevant to this in #307, I haven't had time yet to take it any further but I'll definitely look into this. Unless someone else gets there first! My initial use case would be for comparing public transport journeys e.g. to analyse the impact of traffic throughout a day, or changes to road layouts etc. over longer periods, on journey times/speeds etc. Sounds like someone else has done the research for what I want to be able to do with trajectories, and I need to read it! I have also said that I'll look into map matching of trajectories, which I think is relevant to this too.

**Comment by anitagraser** [comment]
Agreed, it would be great to have these measures for groups of trajectories for both the network-constrained as well as the unconstrained context.
